source/dub/dub.d

```diff
@@ -638,11 +638,11 @@ class Dub {
 		generator.generate(settings);
 	}
 
-	/** Executes tests on the current project.
+	/** Generate project files using the special test runner (`dub test`) configuration.
 
-		Throws an exception, if unittests failed.
+		Any existing project files will be overridden.
 	*/
-	void testProject(GeneratorSettings settings, string config, NativePath custom_main_file)
+	void testProject(string ide, GeneratorSettings settings, string config, NativePath custom_main_file)
```

Comment:
where does the name `ide` come from?

The only part where you use this is in createProjectConfig like
https://github.com/dlang/dub/blob/a72d2e5acbe2f289800d5eab6d5299a27fdc05ef/source/dub/generators/generator.d#L739

so I think you should rather name this `generator_type` to keep consistent with the other APIs.

Additionally I think you should provide a deprecated overload with the old signature that fills this value with `"build"` to make sure DUB API code stays working.